### PR TITLE
feat: migrate webRequest module to NetworkService (Part 7)

### PR DESCRIPTION
**Pull request by zcbenz**
#### Description of Change

Refs https://github.com/electron/electron/issues/15791.
Refs https://github.com/electron/electron/issues/19602.

This PR makes the webRequest module work again.

Lots of related tests are still failing, and I'll fix them in other PRs.

#### Release Notes

Notes: no-notes

### CODE REVIEW

File: shell/browser/api/atom_api_web_request_ns.cc [hunk 1]
```diff
@@ -316,16 +320,29 @@ template <typename Listener, typename Listeners, typename Event>
...
+    // Note that gin treats Function as Dictionary when doing convertions, so we
+    // have to explicitly check if the argument is Function before try to
+    // convert it to Dictionary.
+    if (gin::ConvertFromV8(args->isolate(), arg, &dict)) {
+      dict.Get("urls", &patterns);
+      args->GetNext(&arg);
+    }
+  }
+
+  if (arg.IsEmpty()) {
```
Comment (deepak1556) [inline shell/browser/api/atom_api_web_request_ns.cc]: Maybe its better to move it down and have a single check ?

`if (arg.IsEmpty() || !(gin::ConvertFromV8(args->isolate(), arg, &listener) || arg->IsNull()))`

File: shell/browser/api/atom_api_web_request_ns.cc [hunk 2]
```diff
@@ -395,7 +418,11 @@ void WebRequestNS::OnListenerResult(uint64_t id,
       ReadFromResponse(isolate, &dict, out);
   }
 
-  std::move(callbacks_[id]).Run(result);
+  // The ProxyingURLLoaderFactory expects the callback to be executed
+  // asynchronously, because it used to work on IO thread before NetworkService.
+  base::SequencedTaskRunnerHandle::Get()->PostTask(
```
Comment (deepak1556) [inline shell/browser/api/atom_api_web_request_ns.cc]: What happens when its not executed async ?

Comment (zcbenz) [reply shell/browser/api/atom_api_web_request_ns.cc]: ProxyingURLLoaderFactory's logic is:
Post callback in next tick => pause network request => continue network request when callback is called

If we execute the callback in current tick then it becomes:
Run callback in current tick => continue network request(no-op) => pause network request

So the network request would be paused forever.

Comment (deepak1556) [reply shell/browser/api/atom_api_web_request_ns.cc]: Thanks for the explanation!

**Pull request merged**